util/SerializeUniverse.cpp

```diff
@@ -397,7 +397,7 @@ namespace {
 #else
         int chars_consumed = 0;
         const char* next = buffer.data();
-        auto matched = sscanf(next, "%u%n", &count, &chars_consumed);
+        auto matched = sscanf(next, "%lu%n", &count, &chars_consumed);
```

Comment:
I think this one's debatable. I haven't tried cross-building, for instance for any of the arm 32-bit archs where my change could show a warning. Later down the line one could replace this call with something other than `sscanf`.

Reply:
Probably better to switch `size_t count` to `unsigned int count`. The issue arises because on 32-bit, those are the same, but on 64-bit, `size_t` is instead `unsigned long int`.

Reply:
I replaced `size_t` with `unsigned int`, but was also wondering if not to use `uint32_t`, since it's guaranteed to be 32bit.

Reply:
From what I've read, the `%u` specifically expects an `unsigned int`, not necessarily a `size_t` or `uint32_t`.